Sources/FirebaseFirestore/FirestoreDataConverter.swift

```diff
@@ -109,7 +109,7 @@ internal struct FirestoreDataConverter {
 
       return firebase.firestore.FieldValue.Map(map)
     case is firebase.firestore.FieldValue:
-      return field as! firebase.firestore.FieldValue
+      return field as! firebase.firestore.FieldValue?
```

Comment:
Wait, isn't that changing the semantics of it?
I'm not an expert but wouldn't that now not crash when `field` is nil?

Reply:
Hmm, yes, this is making it optional where as it was non-optional previously. However, the return value is technically an optional (the `default` case is a returning `nil`).

```
S:\SourceCache\thebrowsercompany\swift-firebase\Sources\FirebaseFirestore\FirestoreDataConverter.swift:112:20: warning: treating a forced downcast to 'firebase.firestore.FieldValue' as optional will never produce 'nil'
110 |       return firebase.firestore.FieldValue.Map(map)
111 |     case is firebase.firestore.FieldValue:
112 |       return field as! firebase.firestore.FieldValue
    |              |     | `- note: use 'as?' to perform a conditional downcast to 'firebase.firestore.FieldValue'
    |              |     `- warning: treating a forced downcast to 'firebase.firestore.FieldValue' as optional will never produce 'nil'
    |              `- note: add parentheses around the cast to silence this warning
113 |     default:
114 |       return nil
```

I think that `field as? firebase.firestore.FiledValue` is likely a better spelling. This will shuffle the crash from this location to the callee.

Reply:
We'd never hit this at all in the case where `field` is `nil`, because the `is firebase.firestore.FieldValue` check would fail, right? Any reason not to follow the  'silence' option in the diagnostic by wrapping in parens? (Or writing this as `.some(field as! firebase.firestore.FieldValue)`?)

Reply:
I can see why the readability is impacted by this. I've adjusted the logic here to do a binding on the unwrapping which avoids the duplicate check and cast (which is a u-opt actually!).